src/leo_large_object_get_handler.erl

```diff
@@ -109,27 +109,45 @@ handle_call(stop, _From, State) ->
-                                                      chunk_key = Key,
-                                                      request = Req,
-                                                      metadata = Meta,
-                                                      reference = Ref_1,
-                                                      transport = Transport,
-                                                      socket = Socket}),
-    case Reply of
-        {ok, _Req} ->
-            CacheMeta = #cache_meta{
-                           md5   = Meta#?METADATA.checksum,
-                           mtime = Meta#?METADATA.timestamp,
-                           content_type = leo_mime:guess_mime(Key)},
-            catch leo_cache_api:put_end_tran(Ref_1, Key, CacheMeta, true);
-        _ ->
-            catch leo_cache_api:put_end_tran(Ref_1, Key, undefined, false)
-    end,
+    Reply = case Mode_1 of
+                write ->
+                    Ret = handle_loop(TotalOfChunkedObjs, #req_info{key = Key,
+                                                                    chunk_key = Key,
+                                                                    request = Req,
+                                                                    metadata = Meta,
+                                                                    reference = Ref_1,
+                                                                    transport = Transport,
+                                                                    socket = Socket}),
+                    case Ret of
+                        {ok, _Req} ->
+                            CacheMeta = #cache_meta{
+                                           md5   = Meta#?METADATA.checksum,
+                                           mtime = Meta#?METADATA.timestamp,
+                                           content_type = leo_mime:guess_mime(Key)},
+                            catch leo_cache_api:put_end_tran(Ref_1, Key, CacheMeta, true);
+                        _ ->
+                            catch leo_cache_api:put_end_tran(Ref_1, Key, undefined, false)
+                    end;
+                read ->
+                    TotalSize = Meta#?METADATA.dsize,
+                    Ret_1 = case catch handle_read_loop(TotalSize, #req_info{key = Key,
+                                                                             request = Req,
+                                                                             reference = Ref_1,
+                                                                             transport = Transport,
+                                                                             socket = Socket}) of
+                                {'EXIT', Reason} ->
+                                    {error, Reason};
+                                Ret ->
+                                    Ret
+                            end,
+                    ok = file:close(Ref_1),
```

Comment:
`file:close` should be place at the same module where the correspondig `file:open`  is called